src/components/application_manager/test/commands/mobile/alert_request_test.cc

```diff
@@ -0,0 +1,237 @@
+            static_cast<mobile_apis::Result::eType>(
+                (*result_msg)[am::strings::msg_params][am::strings::result_code]
+                    .asInt()));
+}
+
+TEST_F(AlertRequestTest, Run_MandatoryParametersAreMissed_UNSUCCESS) {
+  CommandPtr command(CreateCommand<AlertRequest>());
+
+  MockAppPtr app(CreateMockApp());
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(app));
+  ON_CALL(*app, hmi_level())
+      .WillByDefault(Return(mobile_apis::HMILevel::HMI_BACKGROUND));
+  ON_CALL(*app, AreCommandLimitsExceeded(_, _)).WillByDefault(Return(false));
+
+  ON_CALL(app_mngr_, GetPolicyHandler())
+      .WillByDefault(
+          ReturnRef(*static_cast<policy::PolicyHandlerInterface*>(NULL)));
+  EXPECT_CALL(*MockMessageHelper::message_helper_mock(),
+              ProcessSoftButtons(_, _, _, _))
+      .WillOnce(Return(mobile_apis::Result::SUCCESS));
+
+  MessageSharedPtr result_msg(CatchMobileCommandResult(CallRun(*command)));
+  EXPECT_EQ(mobile_apis::Result::INVALID_DATA,
+            static_cast<mobile_apis::Result::eType>(
+                (*result_msg)[am::strings::msg_params][am::strings::result_code]
+                    .asInt()));
+}
+
+TEST_F(AlertRequestTest, Run_MandatoryParametersAreInvalid_UNSUCCESS) {
+  MessageSharedPtr msg(CreateMessage(smart_objects::SmartType_Map));
+  (*msg)[am::strings::msg_params][am::strings::alert_text1] = 0;
+  (*msg)[am::strings::msg_params][am::strings::alert_text2] =
+      "invalid\t\nParam";
+
+  CommandPtr command(CreateCommand<AlertRequest>(msg));
+
+  MockAppPtr app(CreateMockApp());
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(app));
+  ON_CALL(*app, hmi_level())
+      .WillByDefault(Return(mobile_apis::HMILevel::HMI_BACKGROUND));
+  ON_CALL(*app, AreCommandLimitsExceeded(_, _)).WillByDefault(Return(false));
+
+  MessageSharedPtr result_msg(CatchMobileCommandResult(CallRun(*command)));
+  EXPECT_EQ(mobile_apis::Result::INVALID_DATA,
+            static_cast<mobile_apis::Result::eType>(
+                (*result_msg)[am::strings::msg_params][am::strings::result_code]
+                    .asInt()));
+}
+
+TEST_F(AlertRequestTest, Run_SUCCESS) {
+  MessageSharedPtr msg(CreateMessage(smart_objects::SmartType_Map));
+  (*msg)[am::strings::msg_params][am::strings::alert_text1] = "alert_text1";
+  (*msg)[am::strings::msg_params][am::strings::alert_text2] = "alert_text2";
+  (*msg)[am::strings::msg_params][am::strings::tts_chunks][0]
+        [am::strings::text] = "alert_text2";
+
+  CommandPtr command(CreateCommand<AlertRequest>(msg));
+
+  MockAppPtr app(CreateMockApp());
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(app));
+  ON_CALL(*app, hmi_level())
+      .WillByDefault(Return(mobile_apis::HMILevel::HMI_BACKGROUND));
+  ON_CALL(*app, AreCommandLimitsExceeded(_, _)).WillByDefault(Return(false));
+
+  ON_CALL(app_mngr_, GetPolicyHandler())
+      .WillByDefault(
+          ReturnRef(*static_cast<policy::PolicyHandlerInterface*>(NULL)));
```

Comment:
@vlantonov, @okozlovlux, this code compiles and runs properly just because calls  `GetPolicyHandler()` method of `ApplicationManager` just once and immediately sends result to `ProcessSoftButtons()` method of `MessageHelper` which in our case mocked. So, as result, it have no chance to affect anything. But of course it's a dirty workaround, so it have to be relapsed with normal mocked `PolicyHandler`.